src/test/java/hanshyn/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,144 @@
+        bookRequestDto.setIsbn(ISBN);
+        bookRequestDto.setPrice(PRICE);
+        bookRequestDto.setDescription(DESCRIPTION);
+        bookRequestDto.setCoverImage(COVER_IMAGE);
+        bookRequestDto.setCategoryIds(Set.of());
+
+        BookDto expected = new BookDto();
+        expected.setTitle(bookRequestDto.getTitle());
+        expected.setAuthor(bookRequestDto.getAuthor());
+        expected.setIsbn(bookRequestDto.getIsbn());
+        expected.setPrice(bookRequestDto.getPrice());
+        expected.setDescription(bookRequestDto.getDescription());
+        expected.setCoverImage(bookRequestDto.getCoverImage());
+        expected.setCategoryIds(bookRequestDto.getCategoryIds());
+
+        String jsonRequest = objectMapper.writeValueAsString(bookRequestDto);
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/books")
+                .content(jsonRequest)
+                .contentType(MediaType.APPLICATION_JSON)
+        ).andExpect(status().isCreated())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class);
+
+        EqualsBuilder.reflectionEquals(expected, actual,"id");
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Get all books")
+    @Sql(scripts = "classpath:database/controller/delete-all-books.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/controller/add-three-books.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    public void getAll_ReturnListBooks() throws Exception {
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders
+                        .get("/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        List<BookDto> actual = List.of(
+                objectMapper.readValue(result.getResponse().getContentAsString(),
+                        BookDto[].class));
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(EXPECTED_THREE_BOOKS, actual.size());
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Return book by id")
+    @Sql(scripts = "classpath:database/controller/delete-all-books.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/controller/add-three-books.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    public void getBookById_ValidId_ReturnBookDto() throws Exception {
+        BookDto expected = new BookDto();
+        expected.setId(BOOK_ID);
+        expected.setTitle(TITLE);
+        expected.setAuthor(AUTHOR);
+        expected.setIsbn(ISBN);
+        expected.setPrice(PRICE);
+        expected.setDescription(DESCRIPTION);
+        expected.setCoverImage(COVER_IMAGE);
+        expected.setCategoryIds(Set.of());
```

Comment:
Please extract sample objects creation to separate private methods to make test method more simple